integration-tests/bats/replication.bats

```diff
@@ -58,6 +58,32 @@ teardown() {
     [[ "$output" =~ "t1" ]] || false
 }
 
+@test "replication: push branch delete" {
+    cd repo1
+    dolt checkout feature
+    dolt config --local --add sqlserver.global.dolt_replicate_to_remote backup1
+    dolt sql -q "create table t1 (a int primary key)"
+    dolt sql -q "select dolt_commit('-am', 'cm')"
+
+    cd ..
+    dolt clone file://./bac1 repo2
+    cd repo2
+    dolt checkout feature
+    run dolt ls
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "t1" ]] || false
+
+    cd ../repo1
+    dolt checkout main
+    dolt sql -q "call dolt_branch('-df', 'feature');"
+
+    cd ..
+    dolt clone file://./bac1 repo3
```

Comment:
I think you may need an additional test, potentially skipped, where the branch already exists on the replica.

Reply:
added more tests